client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -213,3 +217,51 @@ describe( 'getAllPluginsIndexedBySiteId', () => {
 		expect( pluginsOne ).toBe( pluginsTwo );
 	} );
 } );
+
+describe( 'getFilteredAndSortedPlugins', () => {
+	test( 'Should get an empty array if the requested site is not in the current state', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ nonExistingSiteId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get an empty array if the plugins for this site are still being requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteThreeId ], undefined );
+		expect( plugins ).toHaveLength( 0 );
+	} );
+
+	test( 'Should get a plugin list of length 3 if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		expect( plugins ).toHaveLength( 3 );
+	} );
+
+	test( 'Should get a plugin list containing jetpack if both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], undefined );
+		const siteWithPlugin = {
+			[ siteTwoId ]: {
+				active: jetpack.active,
+				autoupdate: jetpack.autoupdate,
+				update: jetpack.update,
+				version: jetpack.version,
+			},
+		};
+		const { active, autoupdate, update, version, ...pluginProperties } = jetpack;
+		expect( plugins ).toEqual(
+			expect.arrayContaining( [ { ...pluginProperties, sites: siteWithPlugin } ] )
+		);
+	} );
+
+	test( 'Should get a plugin list of length 2 if only site 1 is requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId ], undefined );
+		expect( plugins ).toHaveLength( 2 );
+	} );
+
+	test( 'Should get a plugin list of length 2 if active plugins on both sites are requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId, siteTwoId ], 'active' );
+		expect( plugins ).toHaveLength( 2 );
+	} );
+
+	test( 'Should get a plugin list of length 1 if inactive plugins on site 1 is requested', () => {
+		const plugins = getFilteredAndSortedPlugins( state, [ siteOneId ], 'inactive' );
+		expect( plugins ).toHaveLength( 1 );
+	} );
+} );
```

Comment:
Do we also need some test cases for the sorting?

Reply:
I decided to change the test that was looking only for the Jetpack plugin to look for all of them in sorted order to account for this.

Reply:
I'd say that a single test should not test multiple traits/behaviors. That's why they're called unit tests - because each test tests one unit. This is a minor issue though and not a blocker.